### DISCUSSION TITLE
Images order + Shuffle

### DISCUSSION THREAD

**Discussion by Majakhtar** [nftchef/art-engine#97] [Help + Q&A] [answered]
Hey,

I would like to ask about the order the images are produced. 

My images are generated via shuffle except when they are placed in the folders (images & json) they go back to numerical order, this makes it difficult to know which image was created first and so on. 
I would need to know the correct order for the provenance hash right?
Or am I missing something?
Thank you! keep up the amazing work. xx

**Comment by nftchef** [comment, marked as answer]
The Image and json _names_ are always created sequentially (in the order they are numbered), and (in 99% of the cases) minted sequentially in the smart contract.
The generator will randomly pick the art, making each piece random and unique, even though the number is sequential. 

The provenance hash is generated based on the output order.

**Reply by Majakhtar** [reply]
Thank you =)
